services/api-server/src/simcore_service_api_server/api/errors/http_error.py

```diff
@@ -1,7 +1,31 @@
+from typing import Callable, Type
+
 from fastapi import HTTPException
+from fastapi.encoders import jsonable_encoder
 from starlette.requests import Request
 from starlette.responses import JSONResponse
 
 
 async def http_error_handler(_: Request, exc: HTTPException) -> JSONResponse:
-    return JSONResponse({"errors": [exc.detail]}, status_code=exc.status_code)
+    return JSONResponse(
+        content=jsonable_encoder({"errors": [exc.detail]}), status_code=exc.status_code
+    )
+
+
+def make_http_error_handler_for_exception(
+    status_code: int, exception_cls: Type[BaseException]
+) -> Callable:
```

Comment:
is this what is missing in the catalog/director-v2 to ensure all responses are JSON-based?

Reply:
I thought I added it in the director but now I see that it was in the catalog. This is definitively a function for the new ``fastapi-servicelib`` common package.